### PR TITLE
Fix forced backlogged not being limited by BACKLOG_DAYS

### PR DESCRIPTION
**Pull request by fernandog**
as @ratoaq2 pointed

dividing integers will result in integer so
`self.cycleTime = app.BACKLOG_FREQUENCY / 60 / 24` will result 0.5 but value will be 0

so a backlog of 12 hours, will always have a cycleTime of  0 in case of the default of 720

So when checking if we should run a forced search, it will never reach that limit code:
`if not which_shows and not ((curDate - self._last_backlog) >= self.cycleTime):`
as `(curDate - self._last_backlog)` will always be higher than cycleTime

Imo this code was done to check if user runned a backlog before waiting for next backlog (so, a forced backlog)

image

### CODE REVIEW

**Comment by duramato** [comment]
What is the current behaviour?
If i'm reading correctly, now it will be: episodes with WANTED status older than "Backlog Days" won't ever get searched without user interaction?

**Comment by fernandog** [comment]
How ep become wanted? How long is wanted?

**Comment by duramato** [comment]
Became wanted becouse it aired, let's say it has been WANTED for longer than the "backlog days setting" value, what is the current behaviour and what will be the behaviour with this PR?

**Comment by fernandog** [comment]
I changed this PR and now it only fixed issue described in PR body:

Also added a note in the UI:
image

File: medusa/search/backlog.py [hunk 1]
```diff
@@ -49,6 +49,7 @@ def __init__(self):
         self.amActive = False
         self.amPaused = False
         self.amWaiting = False
+        self.force = False
```
Comment (medariox) [inline medusa/search/backlog.py]: Why do you make a new attribute for this? Just pass it to self.search_backlog(force)

File: medusa/search/backlog.py [hunk 2]
```diff
@@ -90,9 +91,11 @@ def search_backlog(self, which_shows=None):
         curDate = datetime.date.today().toordinal()
         from_date = datetime.date.fromordinal(1)
 
-        if not which_shows and not ((curDate - self._last_backlog) >= self.cycleTime):
-            logger.log(u"Running limited backlog on missed episodes " + str(app.BACKLOG_DAYS) + " day(s) and older only")
+        if not which_shows and self.force:
+            logger.log(u"Running limited backlog search on missed episodes from last {0} days".format(app.BACKLOG_DAYS))
```
Comment (medariox) [inline medusa/search/backlog.py]: Use single quotes

File: medusa/search/backlog.py (hunk 1 above)
Comment (fernandog) [inline medusa/search/backlog.py]: because force is also a attribute of the backlog.

Comment (medariox) [inline medusa/search/backlog.py]: What do you mean?

**Comment by p0psicles** [comment]
If I run a python interpreter and run 4 / 0.5 / 3 it results in 2.666666666...?

**Comment by labrys** [comment]
@p0psicles you have a float in the division thus the result is a float.  Same if you did any of the following:
```python
>>> float(3)/4
0.75

>>> 3.0/4
0.75

>>> 3/4.0
0.75

>>> from  __future__ import division
>>> 3/4  # after future import
0.75
>>> 3 // 4  # for integer division after future import
0
```

**Comment by p0psicles** [comment]
aah yeah, now I see what he meant. well then if you expect a float, just divide by 24.0 right?

**Comment by labrys** [comment]
You should actually promote first element to avoid this kind of issue:
```python
>>> 3 / 4 / 0.5
0

>>> 3.0 / 4 / 0.5
1.5
```

**Comment by labrys** [comment]
Personally I prefer to use `__future__ division` everywhere and use `//` if I need explicit integer division

**Comment by labrys** [comment]
Regardless integer division isnt actually a bug here.
It's being used at L93 in backlog.py as follows:
```
        if not which_shows and not ((curDate - self._last_backlog) >= self.cycleTime):
```
And curDate is a date (min resolution is a day) so 0 is the correct answer for a backlog frequency.

File: medusa/search/backlog.py (hunk 1 above)
Comment (labrys) [inline medusa/search/backlog.py]: If you mean `force` as an attribute (meaning this backlog was triggered as the result of a "forced backlog" action) then the attribute name should be called `forced`.  But as @medariox asked: What do you mean?

Comment (fernandog) [inline medusa/search/backlog.py]: @labrys Yes. Forced because user clicked the force button on manage searches. I also added a note there

**Comment by fernandog** [comment]
@labrys the forced is better than do all that math right? Its what the feature is called "forced search" will use backlog days

**Comment by fernandog** [comment]
@labrys about frequency, in server status it show cycle time of 12 hours

**Comment by fernandog** [comment]
@labrys changed to `forced` and promoted to float. btw he backlog_days limit doesn't use that cycleTime anymore

**Comment by fernandog** [comment]
Another PR ready waiting for approval

File: medusa/search/backlog.py (hunk 1 above)
Comment (fernandog) [inline medusa/search/backlog.py]: @medariox can you accept changes please?

**Pull request merged**